core/src/main/java/org/apache/iceberg/SnapshotSummary.java

```diff
@@ -181,14 +181,14 @@ public Map<String, String> build() {
       builder.putAll(properties);
 
       metrics.addTo(builder);
-      setIf(deletedDuplicateFiles > 0, builder, DELETED_DUPLICATE_FILES, deletedDuplicateFiles);
+      SnapshotSummary.set(builder, DELETED_DUPLICATE_FILES, deletedDuplicateFiles);
```

Comment:
[minor] can call set here as we are in same class `SnapshotSummary`
```suggestion
      set(builder, DELETED_DUPLICATE_FILES, deletedDuplicateFiles);
```

Reply:
Unfortunately non static set is defined and has different signature so SnapshotSummary is necessary
```
public void set(String property, String value) {
      properties.put(property, value);
    }
```

Reply:
I think it probably makes more sense to just call `set` directly. There's no need for a static method anymore.